src/test/java/io/wisetime/connector/patrawin/PatrawinConnectorPerformTagUpdateTest.java

```diff
@@ -4,40 +4,256 @@
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 /**
  * @author shane.xie@practiceinsight.io
+ * @author galya.bogdanova@m.practiceinsight.io
  */
 public class PatrawinConnectorPerformTagUpdateTest {
 
+  private static final int TAG_UPSERT_BATCH_SIZE = 100;
+  private static final String TAG_UPSERT_PATH = "/test/path/";
+
+  private static RandomDataGenerator randomDataGenerator = new RandomDataGenerator();
+  private static PatrawinConnector connector;
+
+  private static PatrawinDao patrawinDao = mock(PatrawinDao.class);
+  private static ApiClient apiClient = mock(ApiClient.class);
+  private static ConnectorStore connectorStore = mock(ConnectorStore.class);
+  private static SyncStore syncStore = mock(SyncStore.class);
+
+  @BeforeAll
+  static void setUp() {
+    RuntimeConfig.setProperty(PatrawinConnectorConfigKey.TAG_UPSERT_BATCH_SIZE, String.valueOf(TAG_UPSERT_BATCH_SIZE));
+    RuntimeConfig.setProperty(PatrawinConnectorConfigKey.TAG_UPSERT_PATH, TAG_UPSERT_PATH);
+    RuntimeConfig.clearProperty(ConnectorConfigKey.CALLER_KEY);
+
+    assertThat(RuntimeConfig.getString(ConnectorConfigKey.CALLER_KEY))
+        .as("CALLER_KEY empty value expected")
+        .isNotPresent();
+
+    assertThat(RuntimeConfig.getInt(PatrawinConnectorConfigKey.TAG_UPSERT_BATCH_SIZE))
+        .as("TAG_UPSERT_BATCH_SIZE should be set to " + TAG_UPSERT_BATCH_SIZE)
+        .contains(TAG_UPSERT_BATCH_SIZE);
+
+    connector = spy(
+        Guice.createInjector(binder -> binder.bind(PatrawinDao.class).toProvider(() -> patrawinDao))
+            .getInstance(PatrawinConnector.class)
+    );
+
+    // Ensure PatrawinConnector#init will not fail
+    doReturn(true)
+        .when(patrawinDao)
+        .hasExpectedSchema();
+
+    doReturn(syncStore)
+        .when(connector)
+        .getSyncStore(any(ConnectorStore.class));
+
+    connector.init(new ConnectorModule(apiClient, connectorStore));
+  }
+
+  @AfterAll
+  static void tearDown() {
+    RuntimeConfig.clearProperty(PatrawinConnectorConfigKey.TAG_UPSERT_BATCH_SIZE);
+    RuntimeConfig.clearProperty(PatrawinConnectorConfigKey.TAG_UPSERT_PATH);
+
+    assertThat(RuntimeConfig.getInt(PatrawinConnectorConfigKey.TAG_UPSERT_BATCH_SIZE))
+        .as("TAG_UPSERT_BATCH_SIZE empty result expected")
+        .isNotPresent();
+    assertThat(RuntimeConfig.getString(PatrawinConnectorConfigKey.TAG_UPSERT_PATH))
+        .isNotPresent();
+  }
+
+  @BeforeEach
+  void setUpTest() {
+    reset(patrawinDao);
+    reset(apiClient);
+    reset(connectorStore);
+    reset(syncStore);
+  }
+
   @Test
-  void syncCases_none_found() {
-    // TODO
+  void syncCases_none_found() throws IOException {
+    when(patrawinDao.findCasesOrderedByCreationTime(any(), anyList(), anyInt()))
+        .thenReturn(ImmutableList.of());
+
+    connector.performTagUpdate();
```

Comment:
```suggestion
    connector.syncCases();
```

And lets verify return value as well.